src/optimizer/rule/regex_optimizations.cpp

```diff
@@ -12,17 +14,137 @@ namespace duckdb {
 RegexOptimizationRule::RegexOptimizationRule(ExpressionRewriter &rewriter) : Rule(rewriter) {
 	auto func = make_uniq<FunctionExpressionMatcher>();
 	func->function = make_uniq<SpecificFunctionMatcher>("regexp_matches");
-	func->policy = SetMatcher::Policy::ORDERED;
+	func->policy = SetMatcher::Policy::PARTIAL_ORDERED;
 	func->matchers.push_back(make_uniq<ExpressionMatcher>());
 	func->matchers.push_back(make_uniq<ConstantExpressionMatcher>());
+	func->matchers.push_back(make_uniq<ConstantExpressionMatcher>());
+
 	root = std::move(func);
 }
 
+struct LikeString {
+	bool exists = false;
+	bool escaped = false;
+	string escaped_character = "\\";
+	string like_string = "";
+};
+
+static LikeString GetLikeStringEscaped(duckdb_re2::Regexp *regexp, bool contains = false) {
+	D_ASSERT(regexp->op() == duckdb_re2::kRegexpLiteralString || regexp->op() == duckdb_re2::kRegexpLiteral);
+	LikeString ret;
+	// case insensitivity may be on now, but it can also turn off.
+	if (regexp->op() == duckdb_re2::kRegexpLiteralString) {
+		auto nrunes = (idx_t)regexp->nrunes();
+		auto runes = regexp->runes();
+		for (idx_t i = 0; i < nrunes; i++) {
+			// don't bother trying to convert regex with control characters to like. Let
+			// RE2 handle that
+			if (iscntrl(runes[i])) {
+				ret.exists = false;
+				return ret;
+			}
+			char chr = toascii(runes[i]);
+			// if a character is equal to the escaped character and the like string is already escaped`
+			if (!contains && (chr == '%' || chr == '_' || chr == ret.escaped_character[0])) {
+				ret.escaped = true;
+				ret.like_string += ret.escaped_character;
+			}
+			auto run_as_str {chr};
+			ret.like_string += run_as_str;
+		}
+	} else {
+		auto rune = regexp->rune();
+		if (iscntrl(rune)) {
+			ret.exists = false;
+			return ret;
+		}
+		char chr = toascii(rune);
+		// if a character is equal to the escaped character return that there is no escaped like string.
+		if (!contains && (chr == '%' || chr == '_' || chr == ret.escaped_character[0])) {
+			ret.escaped = true;
```

Comment:
Maybe for simplicity we should just skip the `%` and `_` characters from this optimization for now? It seems like it adds a number of edge cases that complicate this code significantly and I don't think these characters are very common anyway.